### PR TITLE
Refactor ordering community categories

### PR DESCRIPTION
**Pull request by otlaitil**
- Remove invalid SQL injection guard (faulty regex)
- Simplify controller action
- Use AR to generate the mass update query, for enhanced security (generated SQL itself stays the same)
- Add controller tests for the order action

### CODE REVIEW

File: app/controllers/admin/categories_controller.rb [hunk 1]
```diff
@@ -57,23 +57,8 @@ def update
   end
 
   def order
-    sort_priorities = params[:order]
-                      .reject { |o| !o.match /[0-9]+/} #Guard against sql injection
```
Comment (otlaitil) [inline app/controllers/admin/categories_controller.rb]: `.reject { |o| !o.match /[0-9]+/} #Guard against sql injection`

This regex allowed potential SQL injections to pass the `reject`. Correct regex in Ruby would be `/\A[0-9]+\z/` to match the string to contain only numbers.

The category id's are casted to integers with `to_i` later though, in the original code, so no injection _should_ actually be possible. 

I refactored this mainly for clarity and readability.

**Comment by otlaitil** [comment]
@rap1ds what is the status with this?

**Comment by rap1ds** [comment]
@otlaitil Thanks for the ping, and sorry for slow reaction! I'll check this today/tomorrow.

**Comment by rap1ds** [comment]
Thanks @otlaitil ! The code looks good. Tested this on my local dev environment and it seems to work well!

**Comment by otlaitil** [comment]
Great! Thanks @rap1ds 👍

**Pull request merged**